File: javascript/reflex-client.mjs
// Minimal JavaScript client for Mankei Reflex (System One wire format). Node 18+ or browser.
export class Reflex {
  constructor(baseUrl = "http://127.0.0.1:8088", token = "mankei") { this.url = baseUrl.replace(/\/$/, "") + "/v1/reflex"; this.token = token; }
  choice(instructions, options, descriptions = {}) { return { type: "choice", instructions, criteria: Object.fromEntries(options.map((o) => [o, descriptions[o] ?? null])) }; }
  score(instructions, levels) { return { type: "score", instructions, criteria: levels }; }
  yesNo(instructions) { return { type: "noul", instructions }; }
  async decide(state, questions) {
    const r = await fetch(this.url, { method: "POST", headers: { "Content-Type": "application/json", Authorization: `Bearer ${this.token}` }, body: JSON.stringify({ model: "mankei-reflex", state, questions }) });
    if (!r.ok) throw new Error(`Reflex ${r.status}: ${await r.text()}`);
    return (await r.json()).answers;
  }
}

File: python/reflex_client.py
"""Minimal Python client for Mankei Reflex (System One wire format)."""
import json, urllib.request


class Reflex:
    def __init__(self, base_url="http://127.0.0.1:8088", token="mankei", timeout=60):
        self.url, self.token, self.timeout = base_url.rstrip("/") + "/v1/reflex", token, timeout

    @staticmethod
    def choice(instructions, options, descriptions=None):
        return {"type": "choice", "instructions": instructions,
                "criteria": {o: (descriptions or {}).get(o) for o in options}}

    @staticmethod
    def score(instructions, levels):
        return {"type": "score", "instructions": instructions, "criteria": list(levels)}

    @staticmethod
    def yes_no(instructions):
        return {"type": "noul", "instructions": instructions}

    def decide(self, state, questions):
        body = json.dumps({"model": "mankei-reflex", "state": state, "questions": questions}).encode()
        req = urllib.request.Request(self.url, body, {"Content-Type": "application/json", "Authorization": f"Bearer {self.token}"})
        with urllib.request.urlopen(req, timeout=self.timeout) as r:
            return json.load(r)["answers"]
